mobile/lib/src/presentation/pages/settings/settings_page.dart

```diff
@@ -0,0 +1,186 @@
+    BlocProvider.of<SettingsCubit>(context)
+        .setAge(DateTime.now().difference(datePicked).inDays ~/ 365);
+  }
+}
+
+class _SexButton extends StatelessWidget {
+  const _SexButton({
+    Key key,
+    @required this.onSelected,
+    @required this.savedSex,
+    @required this.activeStyle,
+    @required this.defaultStyle,
+  }) : super(key: key);
+
+  final PopupMenuItemSelected<Sex> onSelected;
+  final Sex savedSex;
+  final TextStyle activeStyle;
+  final TextStyle defaultStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return PopupMenuButton<Sex>(
+      child:
+          // ignore: missing_required_param
+          TextButton(
+              child: Text(savedSex.toString().split('.').last),
+              style: ButtonStyle(
+                  foregroundColor: MaterialStateProperty.all(
+                      Theme.of(context).colorScheme.primary))),
+      onSelected: onSelected,
+      itemBuilder: (BuildContext context) => <PopupMenuItem<Sex>>[
+        PopupMenuItem<Sex>(
+          value: Sex.Male,
+          child: Text(
+            'Male',
+            style: savedSex == Sex.Male ? activeStyle : defaultStyle,
+          ),
+        ),
+        PopupMenuItem<Sex>(
+          value: Sex.Female,
+          child: Text(
+            'Female',
+            style: savedSex == Sex.Female ? activeStyle : defaultStyle,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _BoardButton extends StatelessWidget {
+  const _BoardButton({
+    Key key,
+    @required this.onSelected,
+    @required this.savedBoard,
+    @required this.activeStyle,
+    @required this.defaultStyle,
+  }) : super(key: key);
+
+  final PopupMenuItemSelected<AcquisitionBoard> onSelected;
+  final AcquisitionBoard savedBoard;
+  final TextStyle activeStyle;
+  final TextStyle defaultStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return PopupMenuButton<AcquisitionBoard>(
+      child:
+          // ignore: missing_required_param
+          TextButton(
+              child: Text(savedBoard.toString().split('.').last),
+              style: ButtonStyle(
+                  foregroundColor: MaterialStateProperty.all(
+                      Theme.of(context).colorScheme.primary))),
+      onSelected: onSelected,
+      itemBuilder: (BuildContext context) => <PopupMenuItem<AcquisitionBoard>>[
+        PopupMenuItem<AcquisitionBoard>(
+          value: AcquisitionBoard.Cython,
+          child: Text(
+            'Cython',
+            style: savedBoard == AcquisitionBoard.Cython
+                ? activeStyle
+                : defaultStyle,
+          ),
+        ),
+        PopupMenuItem<AcquisitionBoard>(
+          value: AcquisitionBoard.Ganglion,
+          child: Text(
+            'Ganglion',
+            style: savedBoard == AcquisitionBoard.Ganglion
+                ? activeStyle
+                : defaultStyle,
+          ),
+        ),
+      ],
+    );
+  }
+}
```

Comment:
This is too repetitive. Button's behaviour should change dynamically. Imagine if we had 100 settings to handle.

I recommend at least creating a generic implementation.

Reply:
J'voyais pas de situation ou ça serait le cas (avoir 10+ settings). Mais oui, will do.